src/librustc_trans/trans/mir/lvalue.rs

```diff
@@ -116,8 +116,14 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                             LvalueTy::Downcast { adt_def: _, substs: _, variant_index: v } => v,
                         };
                         let discr = discr as u64;
-                        (adt::trans_field_ptr(bcx, &base_repr, tr_base.llval, discr, field.index()),
-                         if common::type_is_sized(tcx, projected_ty.to_ty(tcx)) {
+                        let is_sized = common::type_is_sized(tcx, projected_ty.to_ty(tcx));
```

Comment:
As in llextra is a nullptr if the type is sized? Works for me. I wasn't sure because of the `if type_is_sized { llextra } else { ptr::null_mut() }` in there.